File: src/RummyAi.Domain/Features/GameDto/Models/Game.cs
using RummyAi.Domain.Features.DeckDto.Models;
using RummyAi.Domain.Features.GameDto.Enum;
using RummyAi.Domain.Features.PlayerDto.Model;
using System.Text.Json;

namespace RummyAi.Domain.Features.GameDto.Models;

public class Game
{
    public GameId GameId { get; init; }
    public GameState GameState { get; set; }
    public Stack HiddenStack { get; set; }
    public List<Stack> OpenStack { get; set; }
    public int Turn { get; set; }
    public List<Player> Players { get; set; }
    public Player? CurrentPlayer { get; set; }
    public List<PlayerCards> PlayerCards { get; set; }
    public DateTime Created { get; set; }

    public Game()
    {
        GameId = Guid.NewGuid();
        GameState = GameState.Init;
        Turn = 0;
        Players = [];
        HiddenStack = new();
        OpenStack = new();
        PlayerCards = [];
        Created = DateTime.UtcNow;
    }

    public Game(GameId gameId)
    {
        GameId = gameId;
        GameState = GameState.Init;
        Turn = 0;
        Players = [];
        HiddenStack = new();
        OpenStack = new();
        PlayerCards = [];
        Created = DateTime.UtcNow;
    }

    public int GetNumberOfPlayers()
    {
        return Players.Count;
    }

    public override string ToString()
    {
        return JsonSerializer.Serialize(this);
    }
}

File: src/RummyAi.WebClient/domain/models/Game.ts
import { GameState } from "~/domain/enums/GameState";
import type { GameId } from "./GameId";
import type { PlayerId } from "./PlayerId";
import type { PlayerCards } from "./PlayerCards";
import type { Stack } from "./Stack";
import type { Player } from "./Player";

export class Game {
    gameId: GameId;
    gameState: GameState;
    hiddenStack: Stack;
    openStack: Stack[];
    turn: Number;
    players: PlayerId[];
    currentPlayer: Player | null;
    playerCards: PlayerCards[];
    created: string;

    public constructor(
        gameId: GameId,
        gameState: GameState,
        hiddenStack: Stack,
        openStack: Stack[],
        turn: Number,
        players: PlayerId[],
        currentPlayer: Player,
        playerCards: PlayerCards[],
        created: string) {
        this.gameId = gameId;
        this.gameState = gameState;
        this.hiddenStack = hiddenStack;
        this.openStack = openStack;
        this.turn = turn;
        this.players = players;
        this.currentPlayer = currentPlayer;
        this.playerCards = playerCards;
        this.created = created;
    }
}